crates/lib-core/src/dialects/syntax.rs

```diff
@@ -186,7 +186,14 @@ pub enum SyntaxKind {
     ProcedureName,
     ExportStatement,
     ProcedureParameterList,
+    ProcedureParameter,
+    ParameterName,
     ProcedureStatements,
+    BeginEndBlock,
+    TryBlock,
+    GotoStatement,
+    Label,
+    BatchSeparator,
```

Comment:
Could we flatten these instead of including Syntax types for them just yet? Syntax types are especially useful for writing rules, and most things don't need them. 

You should be able to remove them and the need for them by just removing the NodeMatcher and turning it into a flat matcher

NodeMatcher(Type, Matcher) -> Matcher

Reply:
This PR is based on #1783, so I make the changes first there and once its merged I will rebase this again.